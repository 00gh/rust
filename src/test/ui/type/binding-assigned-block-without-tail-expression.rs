fn main() {
    let x = {
        println!("foo");
        42;
    };
    let y = {};
    let z = {
        "hi";
    };
    println!("{}", x); //~ ERROR E0277
    println!("{}", y); //~ ERROR E0277
    println!("{}", z); //~ ERROR E0277
    let _: i32 = x; //~ ERROR E0308
    let _: i32 = y; //~ ERROR E0308
    let _: i32 = z; //~ ERROR E0308
}
