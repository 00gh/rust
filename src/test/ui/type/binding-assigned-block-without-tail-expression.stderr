error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/binding-assigned-block-without-tail-expression.rs:10:20
   |
LL |     println!("{}", x);
   |                    ^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/binding-assigned-block-without-tail-expression.rs:11:20
   |
LL |     println!("{}", y);
   |                    ^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/binding-assigned-block-without-tail-expression.rs:12:20
   |
LL |     println!("{}", z);
   |                    ^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: this error originates in the macro `$crate::format_args_nl` which comes from the expansion of the macro `println` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> $DIR/binding-assigned-block-without-tail-expression.rs:13:18
   |
LL |     let _: i32 = x;
   |            ---   ^ expected `i32`, found `()`
   |            |
   |            expected due to this
   |
help: remove this semicolon
   |
LL -         42;
LL +         42
   |

error[E0308]: mismatched types
  --> $DIR/binding-assigned-block-without-tail-expression.rs:14:18
   |
LL |     let _: i32 = y;
   |            ---   ^ expected `i32`, found `()`
   |            |
   |            expected due to this
   |
help: this empty block is missing a tail expression
  --> $DIR/binding-assigned-block-without-tail-expression.rs:6:13
   |
LL |     let y = {};
   |             ^^

error[E0308]: mismatched types
  --> $DIR/binding-assigned-block-without-tail-expression.rs:15:18
   |
LL |     let _: i32 = z;
   |            ---   ^ expected `i32`, found `()`
   |            |
   |            expected due to this
   |
help: this block is missing a tail expression
  --> $DIR/binding-assigned-block-without-tail-expression.rs:7:13
   |
LL |       let z = {
   |  _____________^
LL | |         "hi";
LL | |     };
   | |_____^

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
